### DISCUSSION TITLE
WS2812

### DISCUSSION THREAD

**Discussion by lukaszwitkowski** [RavenSystem/esp-homekit-devices#1622] [General]
Hello i use ws2812 with raven and works great , i have ws2812 from china and there is RBG not RGB can i change that somewhere? WS2812

**Comment by RavenSystem** [comment]
Yes, but I forgot to write it in the Wiki. Add `"cm":[0,1,2,3,4]`

**Comment by lukaszwitkowski** [comment]
This is mine code

{
"c":{"b":[{"g":0,"t":5}]},
  "a":[{
    "t":30,
    "ty":8,
    "g":[5,1,24],
    "b":[{"g":0}],
„cm":[0,1,2,3,4],
"nrz" : [ 0.25, 0.6, 1 ]   }]
}

I use WS2811 you thing i need to add something or connect direct to d1 mini
[image](https://user-images.githubusercontent.com/36861888/141436179-06fb54e0-1e66-4a04-98ef-4e616d7fa0d0.png)

**Reply by RavenSystem** [reply]
It is right, but fix quotes.

**Reply by lukaszwitkowski** [reply]
That change when i coppy code . Do you thing i need to add resistor?
I homekit work fone but dont change light.

**Comment by lukaszwitkowski** [comment]
{
"c":{"b":[{"g":0,"t":5}]},
  "a":[{
    "t":30,
    "ty":8,
    "g":[5,1,50],
    "b":[{"g":0}],
"cm":[0,1,2,3,4],
"nrz":[0.25,0.6,1]
  }]
}

with that code work fine but only on 100%-40% when I change under 40% color change for something strange:D
do you thing I need to add resistor?

**Comment by lukaszwitkowski** [comment]
![image](https://user-images.githubusercontent.com/36861888/141449546-08f28ae5-6ce6-49e9-9cd6-d12bfb04869a.jpeg)

**Comment by lukaszwitkowski** [comment]
solution found:D
change glio to GPIO 3 
Working code:
glio 5 button
gpio 3 - ws2811

{
"c":{"b":[{"g":5,"t":5}]},
"a":[{
"t":30,
"ty":8,
"g":[3,1,50],
"b":[{"g":5}],
"cm":[0,1,2,3,4],
"nrz":[0.25,0.6,1]
}]
}
